expression/column.go

```
@@ -0,0 +1,154 @@
+	// If so, this column's name will be the plain sql text.
+	IsAggOrSubq bool
+
+	// Only used for execution.
+	Index int
+}
+
+// Equal implements Expression interface.
+func (col *Column) Equal(expr Expression) bool {
+	if newCol, ok := expr.(*Column); ok {
+		return newCol.FromID == col.FromID && newCol.Position == col.Position
+	}
+	return false
+}
+
+// String implements Stringer interface.
+func (col *Column) String() string {
+	result := col.ColName.L
+	if col.TblName.L != "" {
+		result = col.TblName.L + "." + result
+	}
+	if col.DBName.L != "" {
+		result = col.DBName.L + "." + result
+	}
+	return result
+}
+
+// MarshalJSON implements json.Marshaler interface.
+func (col *Column) MarshalJSON() ([]byte, error) {
+	buffer := bytes.NewBufferString(fmt.Sprintf("\"%s\"", col))
+	return buffer.Bytes(), nil
+}
+
+// GetType implements Expression interface.
+func (col *Column) GetType() *types.FieldType {
+	return col.RetType
+}
+
+// Eval implements Expression interface.
+func (col *Column) Eval(row []types.Datum, _ context.Context) (types.Datum, error) {
+	return row[col.Index], nil
+}
+
+// Clone implements Expression interface.
+func (col *Column) Clone() Expression {
+	newCol := *col
+	return &newCol
+}
+
+// IsCorrelated implements Expression interface.
+func (col *Column) IsCorrelated() bool {
+	return false
+}
+
+// Decorrelate implements Expression interface.
+func (col *Column) Decorrelate(_ Schema) Expression {
+	return col
+}
+
+// HashCode implements Expression interface.
+func (col *Column) HashCode() []byte {
+	var bytes []byte
+	bytes, _ = codec.EncodeKey(bytes, types.NewStringDatum(col.FromID), types.NewIntDatum(int64(col.Position)))
+	return bytes
+}
+
+// ResolveIndices implements Expression interface.
+func (col *Column) ResolveIndices(schema Schema) {
+	col.Index = schema.GetIndex(col)
+	if col.Index == -1 {
+		log.Errorf("Can't find column %s in schema %s", col, schema)
```

Let it panic, otherwise the function will be too ugly.